packages/compass-generative-ai/src/utils/gen-ai-prompt.ts

```diff
@@ -0,0 +1,178 @@
+    '<filter>{}</filter>',
+    '<project>{}</project>',
+    '<sort>{}</sort>',
+    '<skip>0</skip>',
+    '<limit>0</limit>',
+    '<aggregation>[]</aggregation>',
+    'Additional instructions:',
+    '- Only use the aggregation field when the request cannot be represented with the other fields.',
+    '- Do not use the aggregation field if a find query fulfills the objective.',
+    '- If specifying latitude and longitude coordinates, list the longitude first, and then latitude.',
+    `- The current date is ${getCurrentTimeString()}`,
+  ].join('\n');
+}
+
+function buildInstructionsForAggregateQuery() {
+  return [
+    'Reduce prose to the minimum, your output will be parsed by a machine. ' +
+      'You generate MongoDB aggregation pipelines. Provide only the aggregation ' +
+      'pipeline contents in an array in shell syntax, wrapped with XML delimiters as follows:',
+    '<aggregation>[]</aggregation>',
+    'Additional instructions:',
+    '- If specifying latitude and longitude coordinates, list the longitude first, and then latitude.',
+    '- Only pass the contents of the aggregation, no surrounding syntax.',
+    `- The current date is ${getCurrentTimeString()}`,
+  ].join('\n');
+}
+
+export type UserPromptForQueryOptions = {
+  userPrompt: string;
+  databaseName?: string;
+  collectionName?: string;
+  schema?: unknown;
+  sampleDocuments?: unknown[];
+};
+
+function buildUserPromptForQuery({
+  type,
+  userPrompt,
+  databaseName,
+  collectionName,
+  schema,
+  sampleDocuments,
+}: UserPromptForQueryOptions & { type: 'find' | 'aggregate' }): string {
+  const messages = [];
+
+  const queryPrompt = [
+    type === 'find' ? 'Write a query' : 'Generate an aggregation',
+    'that does the following:',
+    `"${userPrompt}"`,
+  ].join(' ');
+
+  if (databaseName) {
+    messages.push(`Database name: "${databaseName}"`);
+  }
+  if (collectionName) {
+    messages.push(`Collection name: "${collectionName}"`);
+  }
+  if (schema) {
+    messages.push(
+      'Schema from a sample of documents from the collection: ```' +
+        JSON.stringify(schema) +
+        '```'
+    );
+  }
+  if (sampleDocuments) {
+    // When attaching the sample documents, we want to ensure that we do not
+    // exceed the token limit. So we try following:
+    // 1. If attaching all the sample documents exceeds then limit, we attach only 1 document.
+    // 2. If attaching 1 document still exceeds the limit, we do not attach any sample documents.
+    const sampleDocumentsStr = JSON.stringify(sampleDocuments);
+    const singleDocumentStr = JSON.stringify(
+      sampleDocuments.slice(0, MIN_SAMPLE_DOCUMENTS)
+    );
+    const promptLengthWithoutSampleDocs =
+      messages.join('\n').length + queryPrompt.length;
+    if (
+      sampleDocumentsStr.length + promptLengthWithoutSampleDocs <=
+      MAX_TOTAL_PROMPT_LENGTH
+    ) {
+      messages.push(
+        'Sample documents from the collection: ```' + sampleDocumentsStr + '```'
```

Comment:
Can we add newline characters to this string? Looks like it's missing that. We should have it around the template ` ``` ` and the document string. Here and below.